compiler/passes/scopeResolve.cpp

```diff
@@ -162,6 +165,36 @@ void scopeResolve() {
     } else if (fn->_this) {
       AggregateType::setCreationStyle(fn->_this->type->symbol, fn);
     }
+
+    // Adjust class type methods for unmanaged/borrowed
+    if (fn->_this && isClass(fn->_this->type) && fDefaultUnmanaged) {
+
+      ArgSymbol* _this = toArgSymbol(fn->_this);
+      Type* t = _this->type;
+      bool ok = isStableClassType(t);
+
+      if (fn->getModule()->modTag == MOD_USER && !ok) {
+        SET_LINENO(fn->_this);
+
+        // Make sure fn->_this->typeExpr exists
+        if (!_this->typeExpr) {
+          _this->typeExpr = new BlockStmt(new SymExpr(t->symbol));
+          parent_insert_help(_this, _this->typeExpr);
+        }
+        // Now adjust _this->typeExpr (whether we just created it or not)
+        Expr* stmt = toArgSymbol(fn->_this)->typeExpr->body.only();
+
+        if (SymExpr* se = toSymExpr(stmt)) {
+          Expr* result = se;
+          if (fWarnUnstable || fDefaultUnmanaged)
+            result = handleUnstableClassType(se);
+          // Amend _this->type
+          fn->_this->type = result->typeInfo();
+        } else {
+          INT_ASSERT(0);
+        }
+      }
+    }
```

Comment:
Please move the above into a separate function over 'fn', ex. adjustMethodsForDefaultUnmanaged(FnSymbol* fn).

Reply:
While there, how about move 'fDefaultUnmanaged' to be the first expression in the initial if-condition ?
That IMO makes it easier to read and faster to execute in the common case.